server/team_matchmaking_service.py

```diff
@@ -0,0 +1,117 @@
+from typing import NamedTuple
+
+from server import GameService
+from server.decorators import with_logger
+from server.lobbyconnection import ClientError
+from server.players import Player
+from server.team_matchmaker.player_party import PlayerParty
+
+MapDescription = NamedTuple('Map', [("id", int), ("name", str), ("path", str)])
+GroupInvite = NamedTuple('GroupInvite', [("inviting", Player), ("invited", Player), ("party", PlayerParty)])
+
+
+@with_logger
+class TeamMatchmakingService:
+    """
+    Service responsible for managing the global team matchmaking. Does grouping, matchmaking, updates statistics, and
+    launches the games.
+    """
+
+    def __init__(self, games_service: GameService):
+        self.game_service = games_service
+        self.player_parties = dict() # player -> current party
+        self._pending_invites = dict() # invited player -> current pending invite to this player
```

Comment:
Same for type annotation as above; though the type of pending invite is unclear.

Do you imply with this that a player can only have a single pending invite? What if they are invited to multiple parties?

Reply:
I do currently imply this to prevent leaking pending invites and overfloading the user with invites.